src/component.ts

```diff
@@ -26,6 +26,20 @@ interface Options<P> {
   shadowRootInit?: ShadowRootInit;
 }
 
+function converter(value: string | null, type?: unknown) {
+  switch (type) {
+    case "legacy":
+      return value === "" ? true : value;
+    case Boolean:
+      return value !== null;
+    case Number:
+      return value === null ? null : Number(value);
```

Comment:
I might consider using `parseFloat` instead of `Number` unless it's a plus to support hexadecimal and binary as number inputs.

```suggestion
      return value === null ? null : parseFloat(value);
```

Reply:
https://github.com/Polymer/lit-element/blob/408b00f72b3c524df9e946a16f494f4aeae7911d/src/lib/updating-element.ts#L165

Reply:
https://github.com/Polymer/lit-element/blob/408b00f72b3c524df9e946a16f494f4aeae7911d/src/lib/updating-element.ts#L165